research/astro.config.js

```diff
@@ -4,10 +4,63 @@ import sitemap from '@astrojs/sitemap'
 import mdx from '@astrojs/mdx'
 import prefetch from '@astrojs/prefetch'
 import compress from 'astro-compress'
+import { rehypeHeadingIds } from '@astrojs/markdown-remark'
+import autolinkHeadings from 'rehype-autolink-headings'
+import { toString } from 'hast-util-to-string'
+import { h } from 'hastscript'
+
+const AnchorLinkIcon = h(
```

Comment:
This file is a little complicated. Can you move plugin-specific code over to a different file?

I had a look at the Astro docs repo to see how they did it, and they took a similar approach.

We could move this into `./plugins/rehypeHeadings` which exports `rehypeHeadingIds` and `autolinkHeadingPlugin`.

![image](https://user-images.githubusercontent.com/17087167/223556996-f431053d-979d-49ce-ac89-0f42d76e4ee0.png)

Hopefully it keeps things simple for new contributors

Reply:
And maybe add a link to the rehype-autolink docs at the top of the file, just so people know where to go if they need to make changes to that file: https://github.com/rehypejs/rehype-autolink-headings